# Bonus: Temperature Analysis I

In [2]:
# Import dependencies
%matplotlib inline
from matplotlib import style
style.use('fivethirtyeight')
import matplotlib.pyplot as plt

In [3]:
import numpy as np
import pandas as pd
import datetime as dt

In [4]:
# Python SQL toolkit and Object Relational Mapper
import sqlalchemy
from sqlalchemy.ext.automap import automap_base
from sqlalchemy.orm import Session 
from sqlalchemy import create_engine, func , inspect

In [5]:
engine = create_engine("sqlite:///Resources/hawaii.sqlite")

In [6]:
# reflect the tables
Base=automap_base()
Base.prepare(engine, reflect=True)

In [7]:
# View all of the classes that automap found
Base.classes.keys()

['measurement', 'station']

In [8]:
Station=Base.classes.station
Measurement=Base.classes.measurement

In [9]:
session=Session(engine)

In [10]:
# Python SQL toolkit and Object Relational Mapper
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy import Column, Integer, String, Float

class HawaiiPrcpTobs(Base):
   __tablename__ = 'prcptobs'
   id = Column(Integer, primary_key = True)
   station = Column(String)
   date = Column(String)
   prcp = Column(Float)
   tobs = Column(Float)

In [11]:
haw_df=pd.read_csv('Resources/hawaii_measurements.csv')
haw_df

station        date  prcp  tobs
0      USC00519397  2010-01-01  0.08    65
1      USC00519397  2010-01-02  0.00    63
2      USC00519397  2010-01-03  0.00    74
3      USC00519397  2010-01-04  0.00    76
4      USC00519397  2010-01-06   NaN    73
...            ...         ...   ...   ...
19545  USC00516128  2017-08-19  0.09    71
19546  USC00516128  2017-08-20   NaN    78
19547  USC00516128  2017-08-21  0.56    76
19548  USC00516128  2017-08-22  0.50    76
19549  USC00516128  2017-08-23  0.45    76

[19550 rows x 4 columns]

In [12]:
# Using either SQLAlchemy or pandas, get the requested data per the README
engine=create_engine('sqlite:///hawaii_measurements.sqlite')
haw_df.to_sql('prcptobs', engine, if_exists='append', index=False)

In [13]:
Base.metadata.create_all(engine)
session=Session(bind=engine)

In [20]:
# Use t-test to determine if difference in means is statistically significant
hm_df=engine.execute('SELECT * FROM prcptobs')
hm_df.fetchall()

[('USC00519397', '2010-01-01', 0.08, 65),
 ('USC00519397', '2010-01-02', 0.0, 63),
 ('USC00519397', '2010-01-03', 0.0, 74),
 ('USC00519397', '2010-01-04', 0.0, 76),
 ('USC00519397', '2010-01-06', None, 73),
 ('USC00519397', '2010-01-07', 0.06, 70),
 ('USC00519397', '2010-01-08', 0.0, 64),
 ('USC00519397', '2010-01-09', 0.0, 68),
 ('USC00519397', '2010-01-10', 0.0, 73),
 ('USC00519397', '2010-01-11', 0.01, 64),
 ('USC00519397', '2010-01-12', 0.0, 61),
 ('USC00519397', '2010-01-14', 0.0, 66),
 ('USC00519397', '2010-01-15', 0.0, 65),
 ('USC00519397', '2010-01-16', 0.0, 68),
 ('USC00519397', '2010-01-17', 0.0, 64),
 ('USC00519397', '2010-01-18', 0.0, 72),
 ('USC00519397', '2010-01-19', 0.0, 66),
 ('USC00519397', '2010-01-20', 0.0, 66),
 ('USC00519397', '2010-01-21', 0.0, 69),
 ('USC00519397', '2010-01-22', 0.0, 67),
 ('USC00519397', '2010-01-23', 0.0, 67),
 ('USC00519397', '2010-01-24', 0.01, 71),
 ('USC00519397', '2010-01-25', 0.0, 67),
 ('USC00519397', '2010-01-26', 0.04, 76),
 ('USC0051

In [14]:
print(haw_df.keys())

Index(['station', 'date', 'prcp', 'tobs'], dtype='object')


In [15]:
haw_df=engine.execute('SELECT station FROM prcptobs ORDER BY station')
haw_df.fetchall()

[('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC00511918',),
 ('USC005119

In [16]:
session.query(HawaiiPrcpTobs.station).group_by(HawaiiPrcpTobs.station).all()

[('USC00511918'),
 ('USC00513117'),
 ('USC00514830'),
 ('USC00516128'),
 ('USC00517948'),
 ('USC00518838'),
 ('USC00519281'),
 ('USC00519397'),
 ('USC00519523')]

In [17]:
session.query(HawaiiPrcpTobs.station,func.max(HawaiiPrcpTobs.tobs)).group_by(HawaiiPrcpTobs.station).all()

[('USC00511918', 87),
 ('USC00513117', 85),
 ('USC00514830', 85),
 ('USC00516128', 84),
 ('USC00517948', 87),
 ('USC00518838', 83),
 ('USC00519281', 85),
 ('USC00519397', 87),
 ('USC00519523', 85)]

Average Temperature in June and December

In [18]:
from scipy import stats
from scipy import mean

In [19]:
avgerage_temp_june=(session.query(func.avg(HawaiiPrcpTobs.tobs))
                  .filter(func.strftime('%m',HawaiiPrcpTobs.date) == '06')
                  .all())
avgerage_temp_june

[(74.94411764705882)]

In [20]:
avgerage_temp_december=(session.query(func.avg(HawaiiPrcpTobs.tobs))
                  .filter(func.strftime('%m',HawaiiPrcpTobs.date) == '12')
                  .all())
avgerage_temp_december

[(71.04152933421226)]

TOBS for the month of June for all Years in the Data Set

In [21]:
june_temperature=(session.query(HawaiiPrcpTobs.date,HawaiiPrcpTobs.tobs)
               .filter(func.strftime('%m',HawaiiPrcpTobs.date) == '06')
               .all())
june_temperature

[('2010-06-01', 78),
 ('2010-06-02', 76),
 ('2010-06-03', 78),
 ('2010-06-04', 76),
 ('2010-06-05', 77),
 ('2010-06-06', 78),
 ('2010-06-07', 77),
 ('2010-06-08', 78),
 ('2010-06-09', 78),
 ('2010-06-10', 79),
 ('2010-06-11', 78),
 ('2010-06-12', 78),
 ('2010-06-13', 78),
 ('2010-06-14', 77),
 ('2010-06-15', 78),
 ('2010-06-16', 78),
 ('2010-06-17', 77),
 ('2010-06-18', 77),
 ('2010-06-19', 82),
 ('2010-06-20', 78),
 ('2010-06-21', 78),
 ('2010-06-22', 78),
 ('2010-06-23', 78),
 ('2010-06-24', 78),
 ('2010-06-25', 77),
 ('2010-06-26', 76),
 ('2010-06-27', 78),
 ('2010-06-28', 78),
 ('2010-06-29', 78),
 ('2010-06-30', 78),
 ('2011-06-01', 77),
 ('2011-06-02', 78),
 ('2011-06-03', 73),
 ('2011-06-04', 70),
 ('2011-06-05', 76),
 ('2011-06-06', 77),
 ('2011-06-07', 77),
 ('2011-06-08', 77),
 ('2011-06-09', 77),
 ('2011-06-10', 78),
 ('2011-06-11', 77),
 ('2011-06-12', 77),
 ('2011-06-13', 78),
 ('2011-06-14', 78),
 ('2011-06-15', 77),
 ('2011-06-17', 78),
 ('2011-06-18', 78),
 ('2011-06-19

December TOBS for all Years in the Data Set

In [22]:
december_temperature=(session.query(HawaiiPrcpTobs.date,HawaiiPrcpTobs.tobs)
               .filter(func.strftime('%m',HawaiiPrcpTobs.date) == '12')
               .all())
december_temperature

[('2010-12-01', 76),
 ('2010-12-03', 74),
 ('2010-12-04', 74),
 ('2010-12-06', 64),
 ('2010-12-07', 64),
 ('2010-12-08', 67),
 ('2010-12-09', 77),
 ('2010-12-10', 66),
 ('2010-12-11', 69),
 ('2010-12-12', 68),
 ('2010-12-13', 68),
 ('2010-12-14', 71),
 ('2010-12-15', 74),
 ('2010-12-16', 67),
 ('2010-12-17', 66),
 ('2010-12-18', 69),
 ('2010-12-19', 71),
 ('2010-12-23', 70),
 ('2010-12-24', 70),
 ('2010-12-26', 74),
 ('2010-12-27', 74),
 ('2010-12-28', 71),
 ('2010-12-29', 75),
 ('2010-12-30', 75),
 ('2010-12-31', 72),
 ('2011-12-01', 69),
 ('2011-12-02', 68),
 ('2011-12-03', 73),
 ('2011-12-04', 74),
 ('2011-12-05', 73),
 ('2011-12-06', 73),
 ('2011-12-07', 73),
 ('2011-12-08', 73),
 ('2011-12-09', 71),
 ('2011-12-10', 73),
 ('2011-12-11', 73),
 ('2011-12-12', 77),
 ('2011-12-13', 72),
 ('2011-12-14', 71),
 ('2011-12-15', 73),
 ('2011-12-16', 70),
 ('2011-12-17', 73),
 ('2011-12-18', 69),
 ('2011-12-19', 74),
 ('2011-12-20', 72),
 ('2011-12-21', 71),
 ('2011-12-22', 72),
 ('2011-12-23

Filtering Out Null Values From June and December TOBS Lists

In [24]:
june_temperature_list = []
for temp in june_temperature:
    if type(temp.tobs) == int:
        june_temperature_list.append(temp.tobs)
        
december_temperature_list = []
for temp in december_temperature:
    if type(temp.tobs) == int:
        december_temperature_list.append(temp.tobs)

Average Temperature in June at All Stations Across All Available Years in The Dataset

In [25]:
mean(june_temperature_list)

<ipython-input-25-40c50f6d1541>:1: DeprecationWarning: scipy.mean is deprecated and will be removed in SciPy 2.0.0, use numpy.mean instead
  mean(june_temperature_list)


74.94411764705882

Average Temperature in December at All Stations Across All 
Available Years in The Dataset

In [26]:
mean(december_temperature_list)

<ipython-input-26-f3fb47d903fd>:1: DeprecationWarning: scipy.mean is deprecated and will be removed in SciPy 2.0.0, use numpy.mean instead
  mean(december_temperature_list)


71.04152933421226

Paired T-Test  :
To determine the difference in June and December average temperature in Honolu,Hawaii
for a time period between 2010 and 2017 Paired t-test was used. The paired t-test is 
used because the two compared samples of temperature observations are related to the same 
location and represent a difference between summer temperature and winter temperature.

There is no statistical significant difference in the mean of june and december temperatures 
in Honolu and Hawaii. I will conclude that this shows a null hypothesis.

In [30]:
stats.ttest_rel(june_temperature_list[0:200],december_temperature_list[0:200])

Ttest_relResult(statistic=21.812993236189637, pvalue=1.1467721770953326e-54)

# Discuss which t-test to use and why and what your results mean.

The value t-statistics is 21.813, with a given degrees of freedom (199),therefore t-statistics can used to calculate the 
p-value.

The p-value is 1.1468e-54, which is far less than the standard thresholds of 0.05 or 0.01, 
so the null hypothesis is rejected and it can be concluded that there is a statistically significant difference between the June temperature and the December temperature in Honolulu, Hawaii.